### ISSUE TITLE
Storage path: No choice

### ISSUE THREAD

**Issue by slimhering** [nextcloud/android#438]
### Actual behaviour
There is only one choice in 'Storage path' which is /storage/emulated/0/Android/data/com.... (the internal memory)

### Expected behaviour
I'd like to have another (second) option to select a path mounted on the external SDcard.
 
### Steps to reproduce
1. CM-Settings -> Apps -> Nextcloud -> Force Stop -> Clear Chache and Data
2. Nextcloud  -> Settings -> Storage Path -> long press on path -> Override -> path to internal memory gets selected...

### Comment ###
Since a while CM "integrates" the SDcard. It kind of "merges" the external SDcard to the internal flash and both flashes presesnt themselves as one towards the user.

### Environment data
Android version:
Android 6.0.1

Device model: 
Samsung S4 mini GT-I9195

Stock or customized system:
CyanogenMod serranoltexx 13.0-20161113-NIGHTLY (Android 6.0.1)

Nextcloud app version:
1.4.0

Nextcloud server version:
10.0.1

**Comment by slimhering** [comment]
ok, I was wrong: The path DOES point to the external SDcard. 

However, I'd like to synchronize eg. the "Music" folder in my home directory with the "Music"-folder on the nectcloud-server... So, I thought this might be achievable by pointing the nextcloud-client-root-directory to my home directory...

Any chance to do so? Thanks,
slimhering

**Comment by AndyScherzinger** [comment]
Hi @slimhering, unfortunately not at the moment. This will be possible in a future version of the new Auto upload which will first be extended to Android 5&6,add video and audio local folders to be auto uploaded to the server, then custom folders, then sync of any of these folders into both directions. 
Hope this gives you an idea of what is to come but these steps are still quite some work :(

**Comment by sasoiliev** [comment]
Hi @AndyScherzinger,

I'd like to question the current behavior of the application regarding the storage path and in particular the use of [Context#getExternalFilesDir()](https://developer.android.com/reference/android/content/Context.html#getExternalFilesDir(java.lang.String)) API. This API 

> Returns the absolute path to the directory on the primary shared/external storage device where the application can place persistent files it owns.

I'd argue that the application does not own the files it syncs. I (the user) own them - that's the primary point of using an application that I can host myself and, as the owner, I'd like to access those files easily (from any application.)

I'd propose to either
* ask for storage path during the account setup (as does the desktop client) or
* use a default path (I'd say `Environment.getExternalStorageDirectory() + File.separator + "NextCloud"` - again, as the desktop client does) and allow the user to select a different folder via some sort of a browse dialog.

As for the instant upload feature that you mention - even though it is the primary concern of @slimhering it is a separate one IMO. What I mean is that it can be accomplished irregardless of the actual storage path, right?

**Comment by mid1221213** [comment]
Hi @sasoiliev, @AndyScherzinger,

I have the same problem of syncing my music to a place where the Music app (from my Z3) will scan it.

While waiting for this, I just had to get the source from git and change the line 68 in DataStorageProvider.java to use the method getExternalMediaDirs() instead of getExternalFilesDirs(null).

Of course, I've made this only for myself and installed the app with adb, but I'd like to know whether this functionality will be included some time in the future. Well, I've seen many people having the same issue as I had, and the solution is simple, again for me, as I don't know what this change really implies ;-)

Disclaimer: I'm not a "real" Android dev :-p
